crates/astria-sequencer/src/action_handler/impls/remove_markets.rs

```diff
@@ -0,0 +1,264 @@
+use astria_core::{
+    connect::market_map::v2::MarketMap,
+    protocol::transaction::v1::action::RemoveMarkets,
+};
+use astria_eyre::eyre::{
+    self,
+    bail,
+    Context as _,
+    OptionExt as _,
+};
+use cnidarium::StateWrite;
+use indexmap::IndexMap;
+
+use crate::{
+    action_handler::ActionHandler,
+    address::StateReadExt as _,
+    app::StateReadExt as _,
+    connect::market_map::state_ext::{
+        StateReadExt as _,
+        StateWriteExt as _,
+    },
+    transaction::StateReadExt as _,
+};
+
+#[async_trait::async_trait]
+impl ActionHandler for RemoveMarkets {
+    async fn check_stateless(&self) -> eyre::Result<()> {
+        Ok(())
+    }
+
+    async fn check_and_execute<S: StateWrite>(&self, mut state: S) -> eyre::Result<()> {
+        // check that the signer of the transaction is a market authority
+        let from = state
+            .try_base_prefixed(
+                &state
+                    .get_transaction_context()
+                    .expect("transaction source must be present in state when executing an action")
+                    .address_bytes(),
+            )
+            .await
+            .wrap_err("failed to convert signer address to base prefixed address")?;
+        let market_authorities = state
+            .get_params()
+            .await?
+            .ok_or_eyre("market map params not found in state")?
+            .market_authorities;
+        if !market_authorities.contains(&from) {
+            bail!("address {from} is not a market authority");
+        }
```

Comment:
```suggestion
        ensure!(
            market_authorities.contains(&from),
            "address {from} is not a market authority"
        );
```

same for other actions